### ISSUE TITLE
Name not defined for control ‘Clear Gradient’ and ‘Add gradient’ present under ‘Gradient’.

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#86]
User Experience:

User would find difficulty to identify the controls if role and name are not defined correctly for controls as they won't get the context of that control and also they won't be able to perform functionality  relevant to that control.

Note: User credentials should NOT be included in the bug.

 

Repro Steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode

4. Navigate to the Left nav and select "Styles" button.

5. Navigate to style page and select "Add variation" button.

6. Navigate to the various controls of the "Add variation" dialog box.

7. Open Developer tool using F12 key and  navigate to controls ‘Clear Gradient’ and ‘Add gradient’ appearing  inside ‘Gradient’ button

8. Verify whether name and role is defined correct or not.

 

Actual Result:

Name and title not defined for control ‘Clear Gradient’ and ‘Add gradient’ appearing  inside ‘Gradient’ button, present under ‘Background’ section. Role also not defined correctly as link. Screen Reader announces as ‘link’.

 

Expected Result:

Name and title should be defined for control as ‘Clear Gradient’ and ‘Add gradient’ appearing  inside ‘Gradient’ button, present under ‘Background’ section. Role should be define as ‘button' for control ‘Clear Gradient’ and ‘Add gradient’.

 

Note:

This issue also exists with Chrome.

Tile should also not defined for ‘Add Gradient’ control.

Similar issue exist throughout the application.

 

MAS Reference:

MAS 4.1.2 - Name, Role, Value

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
![image](https://user-images.githubusercontent.com/2320302/76665123-e193de80-6543-11ea-94b8-a9a071d306b3.png)

**Comment by azaslonov** [comment]
Fixed in 0.1.540.
